src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PENamedTypeSymbol.cs

```diff
@@ -2024,7 +2024,14 @@ internal override UseSiteInfo<AssemblySymbol> GetUseSiteInfo()
 
         protected virtual DiagnosticInfo GetUseSiteDiagnosticImpl()
         {
-            DiagnosticInfo diagnostic = null;
+            // GetCompilerFeatureRequiredDiagnostic depends on UnsupportedCompilerFeature being the highest priority diagnostic, or it will return incorrect
```

Comment:
>GetCompilerFeatureRequiredDiagnostic depends on UnsupportedCompilerFeature being the highest priority diagnostic, or it will return incorrect [](http://example.com/codeflow?start=15&length=141)

It looks like GetUseSiteDiagnosticImpl is overridden by a derived type. Could this put this dependency at risk?
Also, it looks like ```_lazyCachedUseSiteInfo``` can be initialized during symbol creation and we will never get here.

Reply:
> It looks like GetUseSiteDiagnosticImpl is overridden by a derived type. Could this put this dependency at risk?

Currently, no. I could do a similar refactoring to what I did for attribute binding, if you want.